openpgp/packet/ecdh.go

```diff
@@ -1,18 +1,83 @@
 package packet
 
 import (
-	"crypto/ecdsa"
-	"errors"
+	"bytes"
+	"io"
 	"math/big"
+
+	"github.com/keybase/go-crypto/openpgp/ecdh"
+	"github.com/keybase/go-crypto/openpgp/errors"
+	"github.com/keybase/go-crypto/openpgp/s2k"
 )
 
-type ecdhPrivateKey struct {
-	ecdsa.PublicKey
-	x *big.Int
+func ECDHKdfParams(pub *PublicKey) []byte {
+	buf := new(bytes.Buffer)
+	oid := pub.ec.oid
+	buf.WriteByte(byte(len(oid)))
+	buf.Write(oid)
+	buf.WriteByte(18) // ECDH TYPE
+	pub.ecdh.serialize(buf)
+	buf.WriteString("Anonymous Sender    ")
+	buf.Write(pub.Fingerprint[:])
+	return buf.Bytes()
 }
 
-func (e *ecdhPrivateKey) Decrypt(b []byte) ([]byte, error) {
-	// TODO(maxtaco): compute the shared secret, run the KDF and
-	// recover the decrypted shard key.
-	return nil, errors.New("ECDH decrypt unimplemented")
+func decryptKeyECDH(priv *PrivateKey, X, Y *big.Int, C []byte) (out []byte, err error) {
+	ecdhpriv, ok := priv.PrivateKey.(*ecdh.PrivateKey)
+	if !ok {
+		return nil, errors.InvalidArgumentError("bad internal ECDH key")
+	}
+
+	Sx, err := ecdhpriv.DecryptShared(X, Y)
+	if err != nil {
+		return
```

Comment:
Thanks! I've seen the "pascal-like" return value assignment in this repo a few times (I think?) so I kind of followed, but I'm fine with both.